### PR TITLE
Bump moto-ext

### PR DESCRIPTION
**Pull request by viren-nadkarni**
- Bumps moto-ext
- Adds `pytz` which was earlier provided by moto-ext but since has been removed upstream
- Skips assertion of `x-amz-sdk-checksum-algorithm` header. This is a negative test that is being fixed upstream and reverted at a later point.

### CODE REVIEW

**Review by alexrashed — APPROVED** [review]
LGTM! 🚀

File: tests/integration/s3/test_s3.py
Comment (alexrashed) [inline tests/integration/s3/test_s3.py:3923]: This indicates that the latest release of moto still contains regression errors (as you referenced, they are addressed in https://github.com/spulec/moto/pull/5765).

**Review by alexrashed — CHANGES_REQUESTED** [review]
Unfortunately, it seems like there's still an issue in the latest version of moto (https://github.com/spulec/moto/pull/5765), so we will have to wait until this is merged...

File: tests/integration/s3/test_s3.py
Comment (viren-nadkarni) [reply tests/integration/s3/test_s3.py:3923]: @alexrashed This is a negative test case (an extra header is returned when it shouldn't be) and unlikely to break user workflows.

I think we can go safely ahead with merging this PR and avoid another iteration with potentially more breakages.

Comment (alexrashed) [reply tests/integration/s3/test_s3.py:3923]: Since I cannot really determine the severity of this header being present or not, I would like to get @bentsku's feedback (I added him as a reviewer). I'll follow his judgement, but I would also suggest adding a comment linking the PR and noting that this is just temporary and should be remove with the next upgrade.

Comment (viren-nadkarni) [reply tests/integration/s3/test_s3.py:3923]: @bentsku Looping you as the author of the test - please raise any concerns with this change if any.

This change will be reverted with the next moto-ext release.

Comment (bentsku) [reply tests/integration/s3/test_s3.py:3923]: Hi, I need to look deeper, but I don't think this header is ever returned by AWS, even when set in the request. If an algorithm is provided in this request header, the response will return `Checksum{Algorithm}=hash` I think. See https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html

So I suppose this would not break anything client side because it's "just" an extra header never used in response. But it should be fixed in moto and actually never be returned. It should only change the "legacy" provider anyway, as the ASF parser won't pick it up from the moto responses.

**Review by bentsku — APPROVED** [review]
I'm okay with the test cases, we just need to think about remove them later. It only concerns the old provider, and should not break user workflows, I'm not sure any SDK is extremely strict with extra headers (though it has the x-amz prefix).

Or in the worst case, we can always delete this header in the old S3 listener if it causes issues.

**Pull request merged**
